File: HTTP_Server_CLI/scenarios/php/1.2.3-dynamic_100.php
<?php
// @label: 100 dynamic routes
// @group: Dynamic routes
// @competitors: all
// Round-robins through all 100 registered dynamic routes.

$prefixes = ['/user/', '/post/article-', '/api/v1/resource-', '/category/cat-', '/tag/tag-', '/product/sku-', '/order/ord-', '/invoice/inv-', '/review/rev-', '/comment/cmt-'];
for ($i = 11; $i <= 100; $i++) {
   $prefixes[] = "/d{$i}/";
}

$paths = [];
for ($i = 0; $i < 1000; $i++) {
   $idx = $i % count($prefixes);
   $id  = $i % 100;
   $paths[] = $prefixes[$idx] . $id;
}

return [
   'method' => 'GET',
   'paths'  => $paths,
];

File: HTTP_Server_CLI/scenarios/1.2.3-dynamic_100.lua
-- @label: 100 dynamic routes
-- @group: Dynamic routes
-- @competitors: all
-- Round-robins through all 100 registered dynamic routes.

local prefixes = {"/user/", "/post/article-", "/api/v1/resource-", "/category/cat-", "/tag/tag-", "/product/sku-", "/order/ord-", "/invoice/inv-", "/review/rev-", "/comment/cmt-"}
for i = 11, 100 do
   prefixes[#prefixes + 1] = "/d" .. i .. "/"
end

local counter = 0

request = function()
   counter = counter + 1
   local idx = (counter % #prefixes) + 1
   return wrk.format("GET", prefixes[idx] .. (counter % 100))
end
